generic-sensor/resources/generic-sensor-helpers.js

```diff
@@ -62,6 +62,7 @@ function sensor_test(func, name, properties) {
 
 function verifySensorReading(pattern, values, timestamp, isNull) {
   function round(val) {
+    if (typeof(val) === "boolean") return val;
```

Comment:
I don't like this special handling here, but I understand that the proximity sensor is the first case where we are testing a non-numeric property as well. I'd rather do something that just returns `val` if the float conversion below fails.

Reply:
Done.